src/library/types.lisp

```diff
@@ -91,4 +91,9 @@
     ;; This shouldn't be pattern matched against with user code.
     ;;
     ;; See fraction.lisp for more info.
-    (%Fraction Integer Integer)))
+    (%Fraction Integer Integer))
+
+  (define-type (StatefulComputation :state :value)
```

Comment:
This can be in stateful-computation.lisp. Types are only predeclared here when it's needed to avoid circular dependencies.